libats-slick/src/main/scala/com/advancedtelematic/libats/slick/db/BootMigrations.scala

```diff
@@ -6,29 +6,50 @@
 package com.advancedtelematic.libats.slick.db
 
 import com.advancedtelematic.libats.http.BootApp
-import com.typesafe.config.ConfigFactory
 import org.flywaydb.core.Flyway
 import org.slf4j.LoggerFactory
 
-trait BootMigrations {
-  self: BootApp =>
+import scala.concurrent.duration.Duration
+import scala.concurrent.{Await, Future}
+import scala.util.{Failure, Success}
 
-  private val _migrateConfig = ConfigFactory.load()
+trait AsyncMigrations {
+  self: BootApp =>
 
   private val _migrateLog = LoggerFactory.getLogger(this.getClass)
 
-  // Database migrations
-  if (_migrateConfig.getBoolean("database.migrate")) {
-    _migrateLog.info("Running migrations")
+  def migrate: Future[Int] = Future {
+    if (config.getBoolean("database.migrate")) {
+      _migrateLog.info("Running migrations")
+
+      val url = config.getString("database.url")
+      val user = config.getString("database.properties.user")
+      val password = config.getString("database.properties.password")
+
+      val flyway = new Flyway
+      flyway.setDataSource(url, user, password)
 
-    val url = _migrateConfig.getString("database.url")
-    val user = _migrateConfig.getString("database.properties.user")
-    val password = _migrateConfig.getString("database.properties.password")
+      val count = flyway.migrate()
+      _migrateLog.info(s"Ran $count migrations")
 
-    val flyway = new Flyway
-    flyway.setDataSource(url, user, password)
+      count
+    } else
+      0
+  }
+}
+
+trait BootMigrations extends AsyncMigrations {
+  self: BootApp =>
 
-    val count = flyway.migrate()
-    _migrateLog.info(s"Ran $count migrations")
+  if(config.getBoolean("ats.database.asyncMigrations")) {
+    migrate.onComplete {
+      case Success(_) =>
+        log.info("Finished running migrations")
```

Comment:
Was the point of having `migrate` return the count so that we can print it here?